mirascope/openai/prompt.py

```diff
@@ -308,22 +337,33 @@ async def async_extract(self, schema, retries=0):
             openai_tool = OpenAITool.from_model(schema)
             return_tool = False
 
-        completion = await self.async_create(tools=[openai_tool])
+        try:
+            completion = await self.async_create(tools=[openai_tool])
+            self._start_time = extraction_start_time
+        except Exception:
+            self._end_time = datetime.datetime.now().timestamp() * 1000
+            self._start_time = extraction_start_time
+            raise
         try:
             tool = completion.tool
             if tool is None:
+                self._end_time = datetime.datetime.now().timestamp() * 1000
```

Comment:
why do all these end times need to be defined in each conditional. shouldnt end time be when completion is complete?

Reply:
These are all above return statements, so it's right when it completes. The completion also has _start_time and _end_time for when it completes

Reply:
i understand that, but the end time should be set when the completion finishes, not when you hit the conditional

Reply:
Resolved, only completion has it now.